services/ext/service.go

```diff
@@ -636,48 +631,81 @@ func communityToInfo(community *communities.Community) *thirdparty.CommunityInfo
 	}
 }
 
-func (s *Service) FetchCommunityInfo(communityID string) (*thirdparty.CommunityInfo, error) {
-	community, err := s.fetchCommunity(communityID, true)
+func (s *Service) fetchCommunityFromStoreNodes(communityID string) (*thirdparty.CommunityInfo, error) {
+	community, err := s.messenger.FetchCommunity(&protocol.FetchCommunityRequest{
+		CommunityKey:    communityID,
+		TryDatabase:     false,
+		WaitForResponse: true,
+	})
 	if err != nil {
 		return nil, err
 	}
+	return communityToInfo(community), nil
+}
+
+// Fetch latest community from store nodes.
+func (s *Service) FetchCommunityInfo(communityID string) (*thirdparty.CommunityInfo, error) {
+	if s.messenger == nil {
+		return nil, fmt.Errorf("messenger not ready")
+	}
+
+	community, err := s.messenger.FindCommunityInfoFromDB(communityID)
+	if err != nil && err != communities.ErrOrgNotFound {
+		return nil, err
+	}
+
+	// Fetch latest version from store nodes
+	if community == nil || !community.IsControlNode() {
+		return s.fetchCommunityFromStoreNodes(communityID)
+	}
 
 	return communityToInfo(community), nil
 }
 
-func (s *Service) fetchCommunity(communityID string, fetchLatest bool) (*communities.Community, error) {
+// Fetch latest community from store nodes only if any collectibles data is missing.
+func (s *Service) FetchCommunityInfoForCollectibles(communityID string, ids []thirdparty.CollectibleUniqueID) (*thirdparty.CommunityInfo, error) {
 	if s.messenger == nil {
 		return nil, fmt.Errorf("messenger not ready")
 	}
 
-	// Try to fetch metadata from Messenger communities
+	community, err := s.messenger.FindCommunityInfoFromDB(communityID)
+	if err != nil && err != communities.ErrOrgNotFound {
+		return nil, err
+	}
 
-	// TODO: we need the shard information in the collectible to be able to retrieve info for
-	// communities that have specific shards
+	if community == nil {
+		return s.fetchCommunityFromStoreNodes(communityID)
+	}
 
-	if fetchLatest {
-		// Try to fetch the latest version of the Community
-		var shard *shard.Shard = nil // TODO: build this with info from token
-		// NOTE: The community returned by this function will be nil if
-		// the version we have in the DB is the latest available.
-		_, err := s.messenger.FetchCommunity(&protocol.FetchCommunityRequest{
-			CommunityKey:    communityID,
-			Shard:           shard,
-			TryDatabase:     false,
-			WaitForResponse: true,
-		})
+	if community.IsControlNode() {
+		return communityToInfo(community), nil
+	}
+
+	contractIDs := func() map[thirdparty.ContractID]struct{} {
```

Comment:
the struct contains pointers, so it shouldn't be used as key. Please use a `map[string]thirdparty.ContractID` instead and use `id.ContractID.HashKey()` as key

Reply:
Good catch. I assumed `HashKey` implicitly satisfies an interface required for a key.

Reply:
https://github.com/status-im/status-go/compare/539a29a71555e038d51293dbb0a7e8187aa43583..2d3768644b0698b2beb48f582ac939f531d4a28e